### PR TITLE
ihaskell: Update to latest commit, default to Python39

### PR DESCRIPTION
**Pull request by essandess**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by essandess** [comment]
This has a broken `@rpath` that needs fixing.

**Comment by essandess** [comment]
> <s>This has a broken `@rpath` that needs fixing.</s>

Fixed and working. Ready to merge if this builds.

**Comment by essandess** [comment]
This is ready to merge. The only CI issue is a single timeout.

File: devel/ihaskell/Portfile [hunk 1]
```diff
@@ -4,10 +4,13 @@ PortSystem          1.0
 PortGroup           github 1.0
 PortGroup           haskell_stack 1.0
 
-github.setup        gibiansky IHaskell e07b198f07d7dc7a12863f8507439ae3834947e4
+github.setup        gibiansky IHaskell 0f1262d3e710518fd734fbda6f2eba33e476836b
 name                [string tolower ${github.project}]
-version             2021.01.28
+# version from https://github.com/gibiansky/IHaskell/blob/master/ihaskell.cabal
+# with commit date appended
+version             0.10.1.2.20210228
```
Comment (cjones051073) [inline devel/ihaskell/Portfile:11]: I think we should either use an official upstream version, or a date, but not a combination of both. A date is used when upstream does not do proper releases, or if the port specifically is designed to provide a ‘devel’ snapshot version.

Comment (essandess) [reply devel/ihaskell/Portfile]: Done thanks.

**Comment by essandess** [comment]
@cjones051073 May we please merge this?

**Pull request merged**